core-tests/src/test/java/org/mule/runtime/core/privileged/exception/WildcardErrorTypeMatcherTestCase.java

```diff
@@ -0,0 +1,180 @@
+import static org.mule.runtime.core.api.exception.WildcardErrorTypeMatcher.WILDCARD_TOKEN;
+import static org.mule.test.allure.AllureConstants.ErrorHandlingFeature.ERROR_HANDLING;
+import static org.mule.test.allure.AllureConstants.ErrorHandlingFeature.ErrorHandlingStory.ERROR_TYPES;
+
+import org.junit.Test;
+
+import org.mule.runtime.api.component.ComponentIdentifier;
+import org.mule.runtime.api.exception.ErrorTypeRepository;
+import org.mule.runtime.api.message.ErrorType;
+import org.mule.runtime.core.api.exception.AbstractErrorTypeMatcherTestCase;
+import org.mule.runtime.core.api.exception.ErrorTypeMatcher;
+import org.mule.runtime.core.api.exception.WildcardErrorTypeMatcher;
+import org.mule.runtime.core.internal.message.ErrorTypeBuilder;
+
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+
+@Feature(ERROR_HANDLING)
+@Story(ERROR_TYPES)
+public class WildcardErrorTypeMatcherTestCase extends AbstractErrorTypeMatcherTestCase {
+
+  @Test
+  public void anyMatchesAll() {
+    ErrorType mockErrorType = mock(ErrorType.class);
+    when(mockErrorType.getParentErrorType()).thenReturn(transformationErrorType);
+    ErrorTypeMatcher anyMatcher = new WildcardErrorTypeMatcher(anyErrorType);
+
+    assertThat(anyMatcher.match(anyErrorType), is(true));
+    assertThat(anyMatcher.match(transformationErrorType), is(true));
+    assertThat(anyMatcher.match(expressionErrorType), is(true));
+    assertThat(anyMatcher.match(mockErrorType), is(true));
+  }
+
+  @Test
+  public void matchEqual() {
+    ErrorTypeMatcher transformationMatcher = new WildcardErrorTypeMatcher(transformationErrorType);
+
+    assertThat(transformationMatcher.match(transformationErrorType), is(true));
+  }
+
+  @Test
+  public void matchChild() {
+    ComponentIdentifier customTransformerIdentifier =
+        ComponentIdentifier.builder().name("custom").namespace(CORE_NAMESPACE_NAME).build();
+    ErrorTypeRepository errorTypeRepository = muleContext.getErrorTypeRepository();
+    ErrorType customTransformerErrorType = errorTypeRepository.addErrorType(customTransformerIdentifier, transformationErrorType);
+    ErrorTypeMatcher transformationMatcher = new WildcardErrorTypeMatcher(transformationErrorType);
+
+    assertThat(transformationMatcher.match(customTransformerErrorType), is(true));
+  }
+
+  @Test
+  public void doesNotMatchParent() {
+    ErrorTypeMatcher transformationMatcher = new WildcardErrorTypeMatcher(transformationErrorType);
+
+    assertThat(transformationMatcher.match(anyErrorType), is(false));
+  }
+
+  @Test
+  public void doesNotMatchSibling() {
+    ErrorTypeMatcher transformationMatcher = new WildcardErrorTypeMatcher(transformationErrorType);
+
+    assertThat(transformationMatcher.match(expressionErrorType), is(false));
+  }
+
+  @Test
+  public void matchWildcardIdentifier() {
+    ErrorType wildcardErrorType = ErrorTypeBuilder.builder().namespace(CORE_NAMESPACE_NAME).identifier(WILDCARD_TOKEN)
```

Comment:
This mixes the matching and the typing together. Error types represent actual errors that can occur, they are collected and restricted, guaranteeing you cannot throw an error you haven't declared or catch a non existent error.